libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -81,33 +77,46 @@ internal class MapboxTripSession(
         legIndex: Int,
         @RoutesExtra.RoutesUpdateReason reason: String
     ) {
-
-        isOffRoute = false
-        invalidateLatestInstructions()
-        roadObjects = emptyList()
-        routeProgress = null
-
-        updateRouteJob = when (reason) {
+        when (reason) {
```

Comment:
It was never exhaustive but I agree that we should throw an exception instead since we're in full control of the update reason and we need to handle all of them.

Reply:
We used to return a `Job` from `when`. If we added a new option to `RouteExtra` the code wouldn't compile because of an unhandled branch, doesn't it make previous `when` statement exhaustive?

Reply:
We always had an `else` clause that just returned a `null` `Job`. We're throwing an exception now.